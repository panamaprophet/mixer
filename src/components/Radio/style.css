.radio {
	display: flex;
	flex-flow: column;
	cursor: pointer;
}

.label {
    display: flex;
    margin-bottom: 8px;
    align-items: center;
}

.control {
    margin-right: 8px;
    width: 16px;
    height: 16px;
    background: #fff;
    border-radius: 50%;
    box-shadow: inset 0 0 0 2px #232323;
}

.isSelected {
	display: flex;
	align-items: center;
	justify-content: center;
}

.isSelected:after {
	width: 8px;
	height: 8px;
	background: #232323;
	border-radius: 50%;
	content: '';
}