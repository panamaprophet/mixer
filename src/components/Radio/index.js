import React from 'react';
import classnames from 'classnames';

import style from './style.css';


const Radio = ({
    name,
    value: currentValue, 
    values = [],
    onChange = () => {},
}) => (
    <div className={style.radio}>
        {values && values.map(value => (
            <div
                key={value}
                className={style.label}
                onClick={() => onChange(value)}
            >
                <div className={classnames(style.control, (value === currentValue) && style.isSelected)} />
                {value}
            </div>
        ))}
    </div>
);

export default Radio;