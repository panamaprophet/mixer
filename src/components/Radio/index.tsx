import React from 'react';
import classnames from 'classnames';
import {SendParameter, SendParameterType} from '/helpers/entities';
import {SendParamValue} from '/models/sends';
import style from './style.css';


interface Props extends SendParameter {
    onChange: (value: SendParamValue) => void,
}

export const Radio: React.FC<Props> = ({
    type,
    value,
    values = [],
    onChange,
}) => {
    if  (type !== SendParameterType.RADIO) {
        return null;
    }

    const items = values.map(v => (
        <div key={String(v)} className={style.label} onClick={() => onChange(v)}>
            <div className={classnames(style.control, (v === value) && style.isSelected)} />
            <div className={style.title}>{v}</div>
        </div>
    ));

    return (<div className={style.radio}>{items}</div>);
};