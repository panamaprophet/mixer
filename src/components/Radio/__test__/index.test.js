import React from 'react';
import {shallow} from 'enzyme';

import Radio from '../';


const props = {
    name: 'radio-test',
    value: 'one',
    values: ['one', 'two', 'three'],
    onChange: jest.fn(),
};

describe('<Radio />', () => {
    it('should renders without any errors', () => {
        const wrapper = shallow(<Radio {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    describe('active value', () => {
        it('has class isSelected', () => {
            const wrapper = shallow(<Radio {...props} />);
            const result = wrapper.find('.label').at(0).find('.control').hasClass('isSelected');

            expect(result).toBe(true);
        });
    });

    describe('onChange', () => {
        it('should be called on item click', () => {
            const wrapper = shallow(<Radio {...props} />);

            wrapper.find('.label').at(2).simulate('click');

            expect(props.onChange).toBeCalled();
        });
    });
});