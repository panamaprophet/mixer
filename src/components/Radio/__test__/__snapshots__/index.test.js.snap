// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should renders without any errors 1`] = `
<div
  className="radio"
>
  <div
    className="label"
    key="one"
    onClick={[Function]}
  >
    <div
      className="control isSelected"
    />
    one
  </div>
  <div
    className="label"
    key="two"
    onClick={[Function]}
  >
    <div
      className="control"
    />
    two
  </div>
  <div
    className="label"
    key="three"
    onClick={[Function]}
  >
    <div
      className="control"
    />
    three
  </div>
</div>
`;
